### ISSUE TITLE
Clicking on the Apertium logo should return user to the home page

### ISSUE THREAD

**Issue by woswos** [apertium/apertium-html-tools#362]
Conventionally clicking on the logo of a website takes users to the home page. When I click on the Apertium logo, nothing happens. I think this is a small but essential detail in the user interface. I will try to make this change and link PR to this issue. 

Is there a historical reason why the logo is not clickable?

**Comment by sushain97** [comment]
Where would the logo take you? html-tools is a SPA.

**Comment by woswos** [comment]
It should work like the current "cancel" button in Apertium's case. I forgot to mention it.

**Comment by sushain97** [comment]
> It should work like the current "cancel" button in Apertium's case. I forgot to mention it.

🤷‍♂ Usually the logo-link is used as a way to provide a `Home` button. I'm not sure what benefit having both `Cancel` and the logo accomplish the same thing adds but I don't have a strong reason to oppose it either.

**Comment by woswos** [comment]
I think it is really annoying and I still accidentally click the logo to go back. I think it is good to comply with the trends that improve user experience.

Also, IMO, calling the `Go Back` button `Cancel` is not right. AFAIK, `Cancel` buttons are _usually_ used for stoping users from possible accidental/harmful changes or actions they can do. 

Some examples [Situation] -> [Possible Buttons]: 
- "Do you want to delete file X?" -> `Cancel/No` `Yes/Confirm/Ok`
- "Do you want to proceed with the payment?" -> `Cancel/No` `Yes/Confirm/Ok`
- "Do you want to subscribe to the newsletter?" -> `Cancel/No` `Yes/Confirm/Ok`
- "_Operating system file open dialog_" -> `Cancel` `Open`
- "_Buttons at the end of a user profile_" -> `Cancel` `Update Profile`

<img width="1792" alt="Screen Shot 2020-02-24 at 12 39 00 AM" src="https://user-images.githubusercontent.com/10498874/75130786-1972d900-569e-11ea-82bd-1e33445a3900.png">

<img width="1792" alt="Screen Shot 2020-02-24 at 12 40 09 AM" src="https://user-images.githubusercontent.com/10498874/75130811-3ad3c500-569e-11ea-8e13-9e506f5431e0.png">

Even a quick Google or DuckDuckGo search shows red buttons with `X` symbol. There is no single `Cancel` button with `←` that connotes going back to the previous page/view. 

In Apertium's case, `Cancel` button is shown on webpage or document translation pages/views and it is not there to stop an action. I think it would make a lot of sense if a `Cancel` button was shown during the translation process to stop the translation to start a new translation in the same category. For example, clicking the `Cancel` button during a document upload would stop upload/translation and show a new document upload page rather than taking me back to the _home_ page/view, where the text translation section is located.

### tl;dr
IMO, the `Cancel` button should be renamed to something like `Go Back`, `Back`, `Home`, or something that really connotates going back to the _page_ or _view_ that users first encountered.

Also, IMO, all `Cancel` buttons should be in the same or similar place across the website, as you have mentioned in issue #338. To achieve this, **we can completely get rid of the `Cancel` button** and have buttons for each section instead. This also future-proofs us if we have OCR or additional methods in the future and helps us not to worry about where to and how to place a `Cancel` button. Of course, the following design is just a suggestion. Doing this change would automatically resolve issues #338, #358, #359, and #362. 

<img width="1792" alt="Screen Shot 2020-02-24 at 1 26 40 AM copy" src="https://user-images.githubusercontent.com/10498874/75133193-cc940000-56a7-11ea-8212-7f5109155f42.png">

Here are some other articles to get other people's opinions on the button selection issue:
https://uxdesign.cc/the-microcopyist-cancellation-confirmation-conflagration-8a6047a4cf9
https://uxplanet.org/5-essential-ux-rules-for-dialog-design-4de258c22116
https://uxdesign.cc/buttons-placement-and-order-bb1c4abadfcb
https://community.uxmastery.com/t/cancel-button-vs-back-navigation/1325
https://www.nngroup.com/articles/cancel-vs-close/

**Bonus Bug:**
If I translate a webpage and then drag a file into the browser for translation, I end up getting two `Cancel` buttons:
<img width="1568" alt="Screen Shot 2020-02-24 at 12 34 35 AM" src="https://user-images.githubusercontent.com/10498874/75130643-833eb300-569d-11ea-9a60-3bc2a890324b.png">

**Comment by sushain97** [comment]
Yeah, I don't remember exactly why we called it `Cancel` instead of `Back`. I recall there being some bikeshedding but it was a while ago. I prefer the latter but it's a bit painful to implement in terms of breaking all the existing localizations which realistically speaking, are very hard to get fully updated.

A re-design of sorts like the one you suggest might also be a good idea.

**Comment by shrey1608** [comment]
Hey sushain like clicking the logo button means refreshing the page so we can just refresh the page wont it work?

**Comment by shrey1608** [comment]
Hey @sushain97  like clicking the logo button means refreshing the page so we can just refresh the page wont it work?

**Comment by sushain97** [comment]
Refreshing the page instead of just resetting the UI is definitely sub-par UX.

**Comment by shrey1608** [comment]
@sushain97 It will take more time than any other operation but i think it is the accurate thing to do.If there is any other method please tell

**Comment by jonorthwash** [comment]
This is now at least 3 issues, but some decisions need to be made.
* Click on logo to reset UI (imho a good idea and obvious feature to have). 
* Add cancel buttons for uploads, in-process translations, etc. (perhaps some of these things are hard to in fact cancel, but we could at least have the UI immediately let the user do other things and simply ignore the result?)
* Redesign the UI around current "canceling", perhaps in part by reconceptualising it as "back", and perhaps in part through some adjustment to the layout.

**Comment by shrey1608** [comment]
- click on the logo and the UI refreshes feature has been done by me i will make PR for it.

- Adding cancel buttons for in-process transaltions would not be useful imo.

@jonorthwash do tell me if i can add something to it and then i will make the PR

**Comment by jonorthwash** [comment]
That sounds reasonable to me, yes.  I think we had some sort of spinner at some point, though? @sushain97?

**Comment by sushain97** [comment]
I think we have a spinner on the whole page during a request. I don't think it's particularly useful to provide a button to cancel an in-flight request since the in-flight request doesn't block any UI interactions. I think we already cancel them automatically if you change what you've typed.

**Comment by shrey1608** [comment]
okay i agree with @sushain97 . So shall i make a PR for just the refresh of the page by clicking the logo?

**Comment by sushain97** [comment]
Again,

```
Refreshing the page instead of just resetting the UI is definitely sub-par UX.
```

**Comment by prsndp** [comment]
Hi @sushain97 , @TinoDidriksen I would like to solve this issue, can you please assign me ?

**Comment by TinoDidriksen** [comment]
@prsndp, ditto as before - just make a PR.

**Comment by 456535688765** [comment]
I have resolved the problem of cancel button

![Screenshot (233)](https://user-images.githubusercontent.com/92295764/224493623-51dcb9af-8915-4ba0-b27c-4e3bf5c2e53c.png)

**Comment by 456535688765** [comment]
Should I make a PR @sushain97

**Comment by sushain97** [comment]
Sorry, what exactly have you solved?

**Comment by 456535688765** [comment]
Before
![Screenshot (235)](https://user-images.githubusercontent.com/92295764/224533405-d254b0d0-e869-4d9e-aecb-318a634c51ad.png)

After resolving the issue

![224493623-51dcb9af-8915-4ba0-b27c-4e3bf5c2e53c](https://user-images.githubusercontent.com/92295764/224533447-e49338c5-5328-41c1-bd71-71c5272d326e.png)

**Comment by sushain97** [comment]
The only difference I see is that the `Cancel` button has moved. As far as I can tell, that does not resolve this issue.

**Comment by iprasandeep** [comment]
Hi! @sushain97, @woswos is right about the button it should be renamed as "back" or "go back". In the current scenario, it is good to call cancel instead of "back". 
Here url cancel button also should be renamed as "back". 

![image](https://user-images.githubusercontent.com/52857138/235224395-f51daf95-813f-4ca1-8076-0c2d983171ae.png)

**Comment by deepdc19** [comment]
Is the issue still open?

**Comment by deepdc19** [comment]
Is anyone currently assigned to the issue?

**Comment by whyrupesh** [comment]
can I take this issue?
